src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs

```diff
@@ -800,6 +810,37 @@ LIMIT 1
         return new GatewayModel.FungibleResourcesCollection(totalCount, previousCursor, nextCursor, items.Take(limit).ToList());
     }
 
+    private async Task<List<ResourceSupplyViewModel>> GetResourcesSupplyData(long[] entityIds, GatewayModel.LedgerState ledgerState, CancellationToken token)
+    {
+        var cd = new CommandDefinition(
+            commandText: @"
+WITH variables (entity_id) AS (SELECT UNNEST(@entityIds))
+SELECT
+    from_state_version as FromStateVersion,
+    resource_entity_id as ResourceEntityId,
+    CAST(total_supply AS text) AS TotalSupply,
+    CAST(total_minted AS text) AS TotalMinted,
+    CAST(total_burned AS text) AS TotalBurned
+FROM variables
+INNER JOIN LATERAL(
+    SELECT from_state_version, resource_entity_id, total_supply, total_minted, total_burned
+    FROM resource_entity_supply_history
+    WHERE from_state_version <= @stateVersion AND resource_entity_id = variables.entity_id
+    ORDER BY from_state_version DESC
+    LIMIT 1
+) resh ON true
+",
+            parameters: new
+            {
+                stateVersion = ledgerState.StateVersion,
+                entityIds = entityIds,
+            },
+            cancellationToken: token);
+
+        var result = (await _dbContext.Database.GetDbConnection().QueryAsync<ResourceSupplyViewModel>(cd)).ToList();
```

Comment:
Why don't we just use EF Core and `.FromSqlInterpolated`? It looks like we could use already existing entity type definitions? Guess this custom `ResourceSupplyViewModel` is needless?